pkg/secrets/secrets.go

```diff
@@ -0,0 +1,104 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"github.com/go-logr/logr"
+	"github.com/redpanda-data/common-go/secrets"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type CloudExpander struct {
+	client secrets.SecretAPI
+	logger *slog.Logger
+}
+
+type ExpanderCloudConfiguration struct {
+	AWSRegion        string
+	AWSRoleARN       string
+	GCPProjectID     string
+	AzureKeyVaultURI string
+}
+
+// NewCloudExpander creates a new CloudExpander
+func NewCloudExpander(
+	ctx context.Context,
+	prefix string,
+	cloudConfig *ExpanderCloudConfiguration,
```

Comment:
nit: Don't accept pointers for structs that are non-optional (and relatively simple).

```go
// Technically valid now but results in a panic
NewCloudExpander(ctx, "", nil)
```

Sadly, go can't distinguish between nil-ability and by value vs by reference.
However, in simple cases you can prevent misuse, accidental or intentional, by accepting `cloudConfig` by value instead of reference.